### PR TITLE
DOC: document read_orc columns order behaviour

### LINKED ISSUE #47944: BUG: Ordering of `columns` is not preserved in `pd.read_orc`
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
In [5]: import cudf

In [6]: df = cudf.DataFrame({"a": [1, 2, 3], "b": ["a", "b", "c"]})

In [7]: df.to_orc("abc")

In [8]: df.to_parquet("abc.parquet")

In [9]: import pandas as pd

In [10]: pd.read_parquet("abc.parquet", columns=['b', 'a'])
Out[10]: 
   b  a
0  a  1
1  b  2
2  c  3

In [11]: pd.read_orc("abc", columns=['b', 'a'])
Out[11]: 
   a  b
0  1  a
1  2  b
2  3  c

In [12]: pd.__version__
Out[12]: '1.4.3'
```

### Issue Description

The ordering of columns in the output is being respected in `pd.read_parquet` but not in `pd.read_orc`. 

### Expected Behavior

The expected behavior is the columns in the output dataframe are in the same order of `columns`

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : e8093ba372f9adfe79439d90fe74b0b5b6dea9d6
python           : 3.9.13.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-76-generic
Version          : #86-Ubuntu SMP Fri Jan 17 17:24:28 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.4.3
numpy            : 1.22.4
pytz             : 2022.1
dateutil         : 2.8.2
setuptools       : 63.3.0
pip              : 22.2.1
Cython           : 0.29.32
pytest           : 7.1.2
hypothesis       : 6.47.1
sphinx           : 5.1.1
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.4.0
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : None
brotli           : 
fastparquet      : None
fsspec           : 2022.7.1
gcsfs            : None
markupsafe       : 2.1.1
matplotlib       : None
numba            : 0.55.2
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 8.0.1
pyreadstat       : None
pyxlsb           : None
s3fs             : 2022.7.1
scipy            : 1.9.0
snappy           : 
sqlalchemy       : 1.4.39
tables           : None
tabulate         : 0.8.10
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None

</details>

### PR DESCRIPTION
**Pull request by markopacak**
- [X] closes #47944
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/io/orc.py [hunk 1]
```diff
@@ -47,6 +47,8 @@ def read_orc(
         ``file://localhost/path/to/table.orc``.
     columns : list, default None
         If not None, only these columns will be read from the file.
+        Output always follows the ordering of the file and not the columns list.
+        This mirrors the original behaviour of :func:`pyarrow.orc.OrcFile.read`.
```
Comment (phofl) [inline pandas/io/orc.py:51]: Can you show how this renders?

Comment (phofl) [reply pandas/io/orc.py:51]: I was thinking about adding ``~`` to the link, could you try that?

Comment (mroeschke) [inline pandas/io/orc.py:51]:
```suggestion
        This mirrors the original behaviour of :external+pyarrow:py:func:`pyarrow.orc.OrcFile.read`.
```

Comment (mroeschke) [reply pandas/io/orc.py:51]: I think https://github.com/pandas-dev/pandas/pull/49709/files#r1024350615 should allow the `read` sphinx link should allow it to render & link to the pyarrow docs

Comment (markopacak) [reply pandas/io/orc.py]: It doesn't create any link (at least in my local environment).

I found this line from the output of `python make.py html`:

    /home/pandas/pandas/io/orc.py:docstring of pandas.io.orc.read_orc:16: WARNING: external py:func reference target not found: pyarrow.orc.OrcFile.read

Maybe I'm missing some configuration?

From a quick search, I found only 2 usages of `external+pyarrow` in the entire project: in [this .rst file](https://github.com/pandas-dev/pandas/blob/main/doc/source/reference/arrays.rst#id77). Their correctly open the respective doc in pyarrow.

Comment (markopacak) [reply pandas/io/orc.py]: Maybe it's worth opening another issue for adding links to all references to pyarrow (only in the docstring).

`pandas.io.orc.to_orc` is also missing `external+pyarrow`.
https://github.com/pandas-dev/pandas/blob/2455f88ba955ba16f3d50521b2e7cf3df5dc1f32/pandas/io/orc.py#L107

Comment (markopacak) [reply pandas/io/orc.py]: @mroeschke @phofl fixed. I had to change `py:func:` to `py:meth:`. Now the link opens correctly

**Review by mroeschke — APPROVED** [review]

File: pandas/io/orc.py (hunk 1 above)
Comment (phofl) [reply pandas/io/orc.py]: Sounds good, could you open an issue?

**Review by phofl — APPROVED** [review]

File: pandas/io/orc.py (hunk 1 above)
Comment (markopacak) [reply pandas/io/orc.py]: I double-checked and they all seem to correctly link to external documentation. 

E.g. [param engine_kwargs for `DataFrame.to_orc`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_orc.html): it has a link to pyarrow, but [in the source-code](https://github.com/pandas-dev/pandas/blob/2455f88ba955ba16f3d50521b2e7cf3df5dc1f32/pandas/io/orc.py#L107) it doesn't have `external+pyarrow` in front.

So we should be good.

**Pull request merged**